site/content/kapp-controller/docs/latest/faq.md

```diff
@@ -38,3 +38,59 @@ Even though the error above is reported, the Package will still be installed and
 should work as expected. It can also still be uninstalled by deleting the
 PackageInstall. The PackageRepository can be recreated and the PackageInstall
 will sync and reconcile without any updates needed to resolve the error.
+
+### How can I generate the valuesSchema from my ytt schema?
+
+If you are using `ytt` as your Package's templating option and have [defined a schema](../../../ytt/docs/latest/how-to-write-schema), you can use `ytt` to generate your `valuesSchema` (which is in OpenAPI v3 format) for you.
+
+This is the recommended workflow:
+
+1. Create an OpenAPI Document from a Data Values Schema file:
+
+    ```bash
+    $ ytt -f schema.yml --data-values-schema-inspect -o openapi-v3 >schema-openapi.yml
+    ```
+   
+    which will produce...
+
+    ```yaml
+    #! schema-openapi.yml
+    openapi: 3.0.0
+    info:
+      version: 1.0.0
+      title: Openapi schema generated from ytt schema
+    paths: {}
+    components:
+      schemas:
+        dataValues:
+          type: object
+          properties:
+            namespace:
+              type: string
+              default: fluent-bit
+    ```
+
+2. Turn your Package CR into a `ytt` template, itself:
+
+    ```yaml
+    #! package.tpl.yml
```

Comment:
this may be too naive of a perspective to be helpful, but i spent a while wondering if "package.tpl" was a new magic ytt invocation before noticing that the line starts with `#!` as opposed to `#@` ... i still don't get where the `tpl` comes from. ToP-Level-domain ?   anyway the unfamiliarity of "tpl" is what made me start wondering if it was a directive and not just a comment indicating the file name.

Reply:
I believe it is short for template, maybe a way to make this clearer is to change it to `#! package-template.yml`. Another option is to leave the file name comments out all together. It is something we add often in ytt docs, but in this context, the file names are quite clear already since there are only three and the command is listed directly below, imo.

Reply:
This is helpful perspective.

I propose we hoist the name _above_ the contents and use the filename @cari-lynn suggests: `package-template.yml`... I'll make that change and we can revisit.